receiver/zipkinreceiver/trace_receiver.go

```diff
@@ -513,6 +513,9 @@ func extractProtoStatus(zs *zipkinmodel.SpanModel) *tracepb.Status {
 	if message == "" && canonicalCodeStr == "" {
 		return nil
 	}
+	zs.Tags["error"] = "true"
```

Comment:
Sorry if this was discussed and I forgot. I thought the we agreed that we won't process tags like this. OC uses status and does not have any special meaning for this tag. So the translation should just translate the zipkin errors to OC status, and later OC > Jaeger translation should correctly translate OC status to how Jaeger represents errors. That should solve this issue.

Reply:
OC status does not carry error boolean flag. Since we did not want to make changes to translators i ensured changes are just made in zipkin-receiver. As you can see in above diff we continue to use OC status for propagating error message.

Reply:
It doesn't need a boolean. It uses status codes to represent errors. Error should be translated to a status code depending on the information present in the span. If we can't extract a specific error from the span but error is still present, we should set the status to a generic status such as "Unknown". 

Once this is set, whatever format the span is exporter in later will have the error properly encoded. (boolean tag for jaeger, status for OC, string tag for zipkin).

Reply:
May be I don't remember it clearly but I think the objection was to not just transform arbitary tags because a span might be exported using jaeger later. 

My understanding of the issue is:

1. Zipkin uses a string `error` attribute to represent errors with the value being the error message.
2. Jaeger uses a boolean `error` attribute to represent the presence of an error and an `error.message` attribute to represent the error details/message.

Currently, this translation does not take place correctly. Is this correct?

If this is the issue then we need fixes in one or probably two places. 

First, zipkin to OC translation should translate a string `error` attribute correctly to OC status. It should do it's best to use the correct OC status code to represent the error and fall-back to using the Unknown status code. If zipkin error attribute it used to set the OC status, it might make sense to remove the error attribute as this would be added back by OC > Zipkin translator when converted back to Zipkin.

Second, OC to Jaeger translation should correctly translate OC Status to a set of `error` and `error.message` attributes. 

This would ensure Zipkin errors get translated correctly to Jaeger errors without adding any special cases in the code like converting error attribute from string to boolean right in zipkin receiver.

Reply:
@owais if you look at my previous PR, i exactly did a change in OC to Jaeger translator. And we had objections in changing OC->Jaeger translation logic as it could potentially break other backends. This is the minimalistic change that we can safely do.

If you are OK with changing translation logic i can add it back. please suggest?

Reply:
@owais @tigrannajaryan Can you review this PR and let me know if we are good to merge?

Reply:
I'm find with the end result but I think we need to change the implementation. Here is what I think will allow us to merge this PR:

1. Do not replace `error` (or any other) attribute with some other value on the source span. We are replacing error attribute on zipkin spans with a boolean values. By doing that, we are already breaking zipkin conventions and making it hard to translate spans correctly. What will happen if someone receives in zipkin and also exports in zipkin? Will they not lose information?

2. Translate zipkin error tags to OpenCensus Status messages. Should be ideally done in Zipkin to OC translator but if is being done in receiver already, then it's fine to add the fix there _if needed_.

3. Check of OC > Jaeger translator correctly translates OC Status to Jaeger error attributes. If not, fix it in OC > Jaeger translator.